app/Http/Controllers/HomeController.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Controllers\Controller;
+use App\Models\User;
+
+class HomeController extends Controller
+{
+
+    public function welcome()
+    {
+        $data = array(
+            'production' => app()->environment('production'),
+            'webpay_plus_credentials_present' => 
+                config('services.transbank.webpay_plus_cc') !=  null && config('services.transbank.webpay_plus_api_key') != null,
+            'webpay_plus_mall_credentials_present' => 
+                config('services.transbank.webpay_plus_mall_cc') != null && config('services.transbank.webpay_plus_mall_api_key') != null,
+            'webpay_plus_deferred_credentials_present' => 
+                config('services.transbank.webpay_plus_deferred_cc') != null && config('services.transbank.webpay_plus_deferred_api_key') != null,
+            'webpay_plus_mall_deferred_credentials_present' => 
+                config('services.transbank.webpay_plus_mall_deferred_cc') != null && config('services.transbank.webpay_plus_mall_deferred_api_key') != null,
+            'oneclick_mall_credentials_present' => 
+                config('services.transbank.oneclick_mall_cc') != null && config('services.transbank.oneclick_mall_api_key') != null,
+            'oneclick_mall_deferred_credentials_present' => 
+                config('services.transbank.oneclick_mall_deferred_cc') != null && config('services.transbank.oneclick_mall_deferred_api_key') != null,
+            'transaccion_completa_credentials_present' => 
+                config('services.transbank.transaccion_completa_cc') != null && config('services.transbank.transaccion_completa_api_key') != null,
+            'transaccion_completa_mall_credentials_present' => 
+                config('services.transbank.transaccion_completa_mall_cc') != null && config('services.transbank.transaccion_completa_mall_api_key') != null,
+            'patpass_comercio_credentials_present' => 
+                config('services.transbank.patpass_comercio_cc') != null && config('services.transbank.patpass_comercio_api_key') != null
+        );
```

Comment:
Quizás me fui en la volá, pero podrías simplificar el tema así: 

Así en welcome.blade.php solo haces un `@if($show_webpay_plus)` y no un `@if (!$data['production'] || $data['production'] && $data['patpass_comercio_credentials_present'])` 
```suggestion
     'show_webpay_plus' => $this->canShow('webpay_plus'),
     'show_webpay_plus_mall' => $this->canShow('webpay_plus_mall'),
     //..
 );
 
 public function canShow($product) {
     if (app()->environment('production') === false) {
         return true;
     }
     
     return config("'services.transbank.{$product}_cc") &&  config("'services.transbank.{$product}_api_key")
 }
```

Reply:
Es solo para simplificar lo más posible la vista, y dejar la pega de saber si se muestra o no en el controller.

Reply:
Entiend porque queda mejor, es necesario que lo haga ahora ? me gustaría pasar esto poducción pronto porque ya me ha tomado harto

Reply:
Lo que diga tu corazón